signer/tests/integration/postgres.rs

```diff
@@ -5580,3 +5580,1200 @@ async fn is_withdrawal_inflight_catches_withdrawals_in_package() {
+    /// "accidentally" override a confirmed rejection event.
+    ///
+    /// This test creates blockchains with the following structure:
+    ///
+    /// ```text
+    ///          ┌────────┐  ┌────────┐  ┌────────┐
+    /// Bitcoin: │   B1   ├──►  B2a   ├──►  B3a   │
+    ///          └─▲──┬───┘  └─▲──────┘  └─▲──────┘
+    ///            ┊  │      ┌─┊──────┐    ┊  The request is confirmed (✔) in S1
+    ///            ┊  └──────► ┊ B2b  │    ┊  and rejected (✖) in both S2a and S2b.
+    ///            ┊         └─┊────▲─┘    ┊  B2b is orphaned by B3a, but B2a is
+    ///            ┊           ┊    ┊      ┊  still confirming the rejection.
+    ///          ┌─┴──────┐  ┌─┴──────┐  ┌─┴──────┐
+    /// Stacks:  │   S1 ✔ ├──►  S2a ✖ ├──►  S3a   │
+    ///          └────┬───┘  └────────┘  └────────┘
+    ///               │      ┌──────┴─┐
+    ///               └─────>│  S2b ✖ │
+    ///                      └────────┘
+    /// ```
+    #[tokio::test]
+    async fn request_with_both_confirmed_and_orphaned_rejection_not_returned() {
+        let db = storage::new_test_database().await;
+
+        let signature_threshold = 2;
+        let min_block_height = 0;
+
+        // Bitcoin blocks:
+        let bitcoin_1 = BitcoinBlock::new_genesis();
+        let bitcoin_2a = bitcoin_1.new_child();
+        let bitcoin_2b = bitcoin_1.new_child();
+        let bitcoin_3a = bitcoin_2a.new_child();
+        // Stacks blocks:
+        let stacks_1 = StacksBlock::new_genesis().anchored_to(&bitcoin_1);
+        let stacks_2a = stacks_1.new_child().anchored_to(&bitcoin_2a);
+        let stacks_2b = stacks_1.new_child().anchored_to(&bitcoin_2b);
+        let stacks_3a = stacks_2a.new_child().anchored_to(&bitcoin_3a);
+
+        // Write our bitcoin + stacks blocks.
+        db.write_blocks(
+            [&bitcoin_1, &bitcoin_2a, &bitcoin_2b, &bitcoin_3a],
+            [&stacks_1, &stacks_2a, &stacks_2b, &stacks_3a],
+        )
+        .await;
+
+        // Get our chain tips.
+        let (bitcoin_chain_tip, stacks_chain_tip) = db.get_chain_tips().await;
+
+        // Assert that the chain tips are what we expect.
+        assert_eq!(bitcoin_chain_tip.as_ref(), &bitcoin_3a.block_hash);
+        assert_eq!(&stacks_chain_tip, &stacks_3a.block_hash);
+
+        // Store a withdrawal request, confirmed in B1/S1.
+        let request = store_withdrawal_request(&db, 1, &bitcoin_1, &stacks_1, &[true, true]).await;
+
+        // We should get the request as pending accepted.
+        let requests = db
+            .get_pending_accepted_withdrawal_requests(
+                bitcoin_chain_tip.as_ref(),
+                &stacks_chain_tip,
+                min_block_height,
+                signature_threshold,
+            )
+            .await
+            .expect("failed to query db");
+        assert_eq!(requests.len(), 1);
+
+        // Reject the withdrawal request in B2a/S2a (canonical).
+        reject_withdrawal_request(&db, &request, &stacks_2a).await;
```

Comment:
nit: can we check that `requests.is_empty()` also right after this? To be sure that rejected because of this and not some strange interaction of the two rejections

Reply:
That case was handled in `request_with_confirmed_rejection_not_returned` so I didn't feel the need to re-test it here. Maybe a comment pointing that out?

Reply:
It was to be extra pedantic in building up the scenario, but it's fine as is; I don't think a comment it's adding value (since it's not code and we need to trust what it says, nothing really relevant for this test itself). Anyway, resolving.